### PR TITLE
Fix editor losing focus after first word on Android.

### LINKED ISSUE #888: Editor lost focus after first word (space bar is pressed) on Android browser.
Having problem with Draft.js editor on Android (Android 6.0 chrome / browser / ) .
whenever I pressed the first space bar, the Editor lost focus. the next space bar is ok. 
Draft.js 0.9.1

### PR DESCRIPTION
**Pull request by tonygentilcore**
This patch narrows the scope of `restoreEditorDOM` to only blow away the
`DraftEditorContents` instead of the entire editor. The key difference is
that the `contentEditable` DOM element is not replaced, which means that
the soft keyboard isn't dismissed on Android.

Manually tested against "Cut" and "IME" tests:
https://github.com/facebook/draft-js/wiki/Manual-Testing#cutcopypaste
https://github.com/facebook/draft-js/wiki/Manual-Testing#ime

Fixes #888, Ref #626, Ref #102

### CODE REVIEW

**Comment by spicyj** [comment]
I'm fine with this. Would be nice to gate it behind a flag when we pull it in initially though in case something breaks.

**Comment by tonygentilcore** [comment]
@spicyj Are you saying you'd like me to update the PR w/ a flag? If so, do you have an example of how you like to do them in this repo?

**Comment by spicyj** [comment]
I think this looks fine as-is and one of us will merge it (and make sure it doesn't break Facebook), more of a note to myself. Thanks for checking!

**Comment by tonygentilcore** [comment]
@spicyj I've changed my project to depend on my fork which contains this fix, but I'd really like to move back to depending on the canonical version. What's the timeline/process for merging this patch?

**Comment by tonygentilcore** [comment]
Friendly ping.

**Comment by Guardiannw** [comment]
Please merge this quickly.  This bug makes Draft.js unusable on the Android web.

**Comment by EthanRutherford** [comment]
Getting this fix in would be greatly appreciated. We're currently having to switch to a fallback editor on mobile devices, and the extra effort of implementing and maintaining code that emulates draft behavior is costly. It would be really great to be able to just use one editor implementation.

**Comment by Guardiannw** [comment]
It looks like this PR still doesn't solve all the problems relating to the android browsers.

**Comment by flarnie** [comment]
Hi @tonygentilcore - I'm picking up testing this internally and merging it.

I'm not able to reproduce the original bug report, in order to verify that this fixes it.
No objectings to merging it, but I need to be able to see that it fixes something.

I'm testing using an android 5.0 device, on various examples of Draft. See the gif - I might be missing something. Please advise about how to reproduce the bug, and thanks again for working on this.

We are starting to incrementally improve Draft for android web, so expect more attention to this in the coming weeks.
cant_repro_android5_bug

**Comment by flarnie** [comment]
PS there is a separate bug, not fixed by this, which seems related to autocorrect. Opened a new issue for that here - https://github.com/facebook/draft-js/issues/1010

**Comment by flarnie** [comment]
I was thinking that this was intended to fix the issue with extra characters appearing that is described in the first comment on #888. And coincidentally, I found that bug does repro on Android 5.1, but this doesn't fix it. :) Opening a separate issue for that bug. https://github.com/facebook/draft-js/issues/1011

So I now assume this is just addressing the issue of the editor "lost focus after the first word" which is the title of #888. But based on the descriptions of that bug, I'm still not able to reproduce it. Have tried on Android 5.0, 5.1, 6.0.1, and 7.1.1. More details or verification would be great.

Basically if someone else can post a gif or swear on their honor that this fixes the bug, and/or let me know what device/osx you able to repro on, then I would be ok merging this.

Thanks to everyone for your patience.

**Comment by flarnie** [comment]
Ok - after asking around I think I understand now how to trigger the bug, and verified that this PR will fix it. Here is what I saw:
1) Enter non-whitespace characters, enter a space, then enter more non-whitespace characters
2) Delete the characters
3) When you hit the space, the keyboard closes.

That is in Android 5.0 and 7.1, and I verified the fix in 5.0, trusting that it works the same way in later versions.

I'm continuing the internal testing of this change, and expect to merge this soon if no problems arise.

**Comment by AndrewSouthpaw** [comment]
Hey @flarnie, thanks for starting work on some of these PRs. I've been working on Android compatibility for Draft JS as well for a few days and done a bunch of research. This PR does address the issue with the keyboard losing focus, but does not fix some other autocomplete behavior. I've opened a new PR #1013 which fixes both problems. Take a look and see if that seems like a more fitting solution. All credit goes to the original author, I just brought it up to date with the recent version of the codebase, because it was abandoned a long time ago. :)

**Comment by aaren-cordova** [comment]
Verified fix on Galaxy S6, Galaxy S7 in Chrome browser.  Keyboard no longer loses focus after non-word char (comma, space, dollar sign, ect.)

**Comment by flarnie** [comment]
Thanks @AndrewSouthpaw and @aaren-cordova for connecting this to the other fix!

#1013 looks pretty exciting! Will be reviewing and testing that, and hope to merge it ASAP. For now going to merge this since we've tested it pretty well and it seems to at least improve some android use cases.

**Comment by abhinavsingi** [comment]
@flarnie When will you release this? Currently package is not published for this fix. Can you please tell an eta on that?
